contracts/src/StakeTableV2.sol

```diff
@@ -290,6 +373,93 @@ contract StakeTableV2 is StakeTable, PausableUpgradeable, AccessControlUpgradeab
+        address validator = msg.sender;
+        ensureValidatorActive(validator);
+        require(newCommission <= 10000, InvalidCommission());
+
+        CommissionTracking storage tracking = commissionTracking[validator];
+        uint16 currentCommission = tracking.commission;
+        uint256 lastIncreaseTime = tracking.lastIncreaseTime;
+
+        require(newCommission != currentCommission, CommissionUnchanged());
+
+        // NOTE: Limits exist to protect stakers from sudden loss of revenue due
+        // to commission increases.
+        //
+        // 1. Limits only enforced for commission *increase*.
+        // 2. Time of last change only tracked for commission *increase*.
+        if (newCommission > currentCommission) {
+            // Allow immediate first increase or after interval has passed
+            require(
+                lastIncreaseTime == 0
+                    || block.timestamp >= lastIncreaseTime + minCommissionUpdateInterval,
+                CommissionUpdateTooSoon()
+            );
+
+            uint16 increase = newCommission - currentCommission;
+            require(increase <= maxCommissionIncrease, CommissionIncreaseExceedsMax());
+            tracking.lastIncreaseTime = block.timestamp;
+        }
+        tracking.commission = newCommission;
+
+        emit CommissionUpdated(validator, block.timestamp, newCommission);
+    }
+
+    /// @notice Set the minimum interval between commission updates
+    /// @param newInterval The new minimum interval in seconds
+    function setMinCommissionUpdateInterval(uint256 newInterval) external virtual onlyOwner {
+        require(newInterval > 0 && newInterval <= 365 days, InvalidRateLimitParameters());
+        minCommissionUpdateInterval = newInterval;
+        emit MinCommissionUpdateIntervalUpdated(newInterval);
+    }
+
+    /// @notice Set the maximum commission increase allowed per update
+    /// @param newMaxIncrease The new maximum increase in basis points (e.g., 500 = 5%)
+    function setMaxCommissionIncrease(uint16 newMaxIncrease) external virtual onlyOwner {
+        require(newMaxIncrease > 0 && newMaxIncrease <= 10000, InvalidRateLimitParameters());
+        maxCommissionIncrease = newMaxIncrease;
+        emit MaxCommissionIncreaseUpdated(newMaxIncrease);
+    }
+
+    /// @notice Initialize validator commissions during V2 migration
+    /// @dev This function is used to retroactively initialize commission storage for validators
+    /// that were registered before the V2 upgrade. On decaf, this will be called with current
+    /// commission values read from L1 events. On mainnet, this will be called with an empty array
+    /// since there are no pre-existing validators.
+    /// @param initialCommissions Array of InitialCommission structs containing validator addresses
+    /// and their commissions
+    function _initializeCommissions(InitialCommission[] calldata initialCommissions) private {
```

Comment:
before we start the for loop, we should do a check to see if the length of initialCommissions is equal to the number of validators in the contract, and revert otherwise

Reply:
Uhm... we can't get the number of validators from the validators mapping and don't store it otherwise.

Reply:
There is a problem though if the validator has zero commission, then after initialization it's not possible to tell if the tracking is initialized or not.